name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DABPDFExtractor.java

```diff
@@ -60,47 +60,55 @@ private void addBuyTransaction()
                         .section("isin", "name", "currency") //
                         .find("Gattungsbezeichnung ISIN") //
                         .match("^(?<name>.*) (?<isin>[^ ]*)$")
-                        .match("STK [\\d.]+(,\\d+)? (?<currency>\\w{3}+) ([\\d.]+,\\d+)$")
+                        .match("STK [\\d\\.]+(,\\d+)? (?<currency>\\w{3}) ([\\d\\.]+,\\d+)$")
```

Comment:
Inside character classes, `.` is not special, so `"[\\d.]"` (as it was) already does what it should do, i.e. match any digit or the dot (but no other characters).

The `+` in `x{3}+` turns the repetition into a possessive quantifier. I don’t think this will change the meaning in any way in this particular case, but removing the `+` might slow down the matching slightly.

In short, I think you should leave those as they were.

Reply:
> The ```+``` in ```x{3}+``` turns the repetition into a possessive quantifier. I don’t think this will change the meaning in any way in this particular case, but removing the ```+``` might slow down the matching slightly.

I did not know about the impact of ```+```. Interesting!
Either way, I think speed is not the problem here (if feels the conversion of the PDF to text uses all the time).
I agree on your first point.

Reply:
Thanks, Ulrich. The timing is funny though, because just today I was reading some more about the performance of regex and how to speed it up. You are right with the `+` and its impact on performance here. In future, I will try to stick to what is suggested here: https://www.loggly.com/blog/five-invaluable-techniques-to-improve-regex-performance/

Reply:
Nice link. thanks.